etc/cmake/options.cmake

```diff
@@ -34,58 +34,126 @@ option(OT_FTD "enable FTD" ON)
 option(OT_MTD "enable MTD" ON)
 option(OT_RCP "enable RCP" ON)
 
-option(OT_ANYCAST_LOCATOR "enable anycast locator support")
-if(OT_ANYCAST_LOCATOR)
-    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_TMF_ANYCAST_LOCATOR_ENABLE=1")
-endif()
+set(OT_CONFIG_VALUES
+    ""
+    "ON"
+    "OFF"
+)
+
+macro(ot_option name ot_config description)
+    # Declare an OT cmake config with `name` mapping to OPENTHREAD_CONFIG
+    # `ot_config`. Parameter `description` provides the help string for this
+    # OT cmake config. There is an optional last parameter which if provided
+    # determines the default value for the cmake config. If not provided
+    # empty string is used which will be treated as "not specified". In this
+    # case, the variable `name` would still be false but the related
+    # OPENTHREAD_CONFIG is not added in `ot-config`.
+
+    if (${ARGC} GREATER 3)
+        set(${name} ${ARGN} CACHE STRING "enable ${description}")
```

Comment:
I think this change will affect on the cmake-gui. Previously, all these options will be represented as check box as follows:

<img width="473" alt="image" src="https://user-images.githubusercontent.com/2240212/182995912-a51d1856-8f04-4a88-8674-abaa8a6b0db7.png">

Reply:
I expect the checkboxes are used for boolean options. Since we need 3 values (on, off and unspecified), we cannot use boolean to realize this. I expect with new model they may be show as multi-choice option?

The challenge with previous model of OT cmake option is that it did not follow a consistent pattern:
- For most configs, we could only enable the feature and setting it to off did not actually disable it but instead left it unspecified. 
- But for some configs (`OT_ASSERT`, OT_CSL_AUTO_SYNC`, `BBR` one, ect) they did not follow this pattern and would disable the feature when set to off.